en/xml/structure/topic.preserve-entities.xml

```diff
@@ -15,32 +15,212 @@
+        <para>The script converts all entities to their protected notation in
+          place and creates backup files with the extension <filename
+            class="extension">.bak</filename>.</para>
+      </step>
+      <step>
+        <para>Process your XML file with your XML tools.</para>
+      </step>
+      <step>
+        <para>Revert all the protected notation to their original entity
+          notation:</para>
+        <screen><prompt>$ </prompt><command>text2ents.py</command> XMLFILE1 XMLFILE2...</screen>
+      </step>
+    </procedure>
   </section>
   <section role="discussion">
     <title>Discussion</title>
-    <para>Some discussion about problem and solution.</para>
+    <para><emphasis>Not</emphasis> expanding entities looks like
+      an abnormal use case as XML parsers are supposed to expand entities by
+      default.
+      However, expanding entities has one disadvantage: after the XML parser
+      has resolved entities, the content is indistinguishable from other
+      content. In other words, you cannot distinguish content that comes
+      from an entity definition from existing content. You cannot
+      <quote>go back</quote> and revert this process once all entities are
+      expanded.
+    </para>
+    <para>
+      For this reason, preserving entities can be useful when you want to
+      process your XML, but keep your entities untouched. <xref
+        linkend="pro.structure.preserve-entities.workflow"
+        xrefstyle="select:label"/> showed one solution. However, you should
+      keep in mind some issues that may or may not lead to problems in your
+      XML files:
+    </para>
+
+    <itemizedlist>
+      <listitem>
+        <para>The script does not know XML. As such, it reads the XML file
+          line by line and replaces each line through regular expressions.</para>
+      </listitem>
+      <listitem>
+        <para>The script can replace entities in situations where it is
+        inappropriate. For example, if you have an entity in an <tag
+          class="attribute">href</tag> of a <tag>xi:include</tag> element,
+          such reference will not work anymore when resolving XIncludes.
+        </para>
+      </listitem>
+      <listitem>
+        <para>The script assumes it does not have to deal with external
+          entities. External entities refer to external storage objects,
+          for example:
+        </para>
+        <programlisting language="xml">&lt;!DOCTYPE book [
+  <emphasis role="bold">&lt;!ENTITY intro SYSTEM "chap-intro.xml"></emphasis>
+]>
+&lt;book>
+  &lt;title>...&lt;/title>
+  <emphasis role="bold">&amp;intro;</emphasis>
+&lt;/book></programlisting>
+        <para>Such external entities usually refers to whole XML structures
+          like chapters, sections etc.
+          Replacing such structures with ordinary text would lead to syntax
+          errors in your XML files.
+          However, nowadays such entities are not seen very often. It should
```

Comment:
*nowadays* is colloquial. However, such entities are not used very often, and they should be replaced by XInludes ~~~anyway~~~